src/components/password/password.spec.tsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import { mount, ReactWrapper } from "enzyme";
+import Password, { PasswordProps, TypeOptions } from "./password.component";
+import { assertStyleMatch } from "../../__spec_helper__/test-utils";
+import { HiddenAriaLive } from "./password.style";
+import Label from "../../__internal__/label";
+import Textbox from "../textbox";
+
+function renderPasswordInput(props: PasswordProps) {
+  return mount(<Password {...props} />);
+}
+
+describe("Password Input", () => {
+  let wrapper: ReactWrapper;
+  let input: ReactWrapper;
+
+  describe("when rendered", () => {
+    it("should have the Textbox component as it's child", () => {
+      wrapper = renderPasswordInput({});
+      expect(wrapper.find(Textbox)).toHaveLength(1);
+    });
+  });
+
+  describe("required", () => {
+    beforeAll(() => {
+      wrapper = renderPasswordInput({ label: "required", required: true });
+    });
+
+    it("the required prop is passed to the input", () => {
+      input = wrapper.find("input").at(0);
+      expect(input.prop("required")).toBe(true);
+    });
+
+    it("the isRequired prop is passed to the label", () => {
+      const label = wrapper.find(Label).at(0);
+      expect(label.prop("isRequired")).toBe(true);
+    });
+  });
+
+  describe("password prop checks", () => {
+    it.each<TypeOptions>(["password", "text"])(
+      "password input type should be correct when type is %s",
+      (inputTypes) => {
+        wrapper = renderPasswordInput({ type: inputTypes });
+        expect(wrapper.find("input").at(0).prop("type")).toBe(inputTypes);
+      }
+    );
+
+    it("Password input id should be correct when id is passed as a custom string", () => {
+      wrapper = renderPasswordInput({ id: "foo" });
+      expect(wrapper.find("input").at(0).prop("id")).toBe("foo");
+    });
+  });
```

Comment:
It might be a good idea to include an additional test for the other case where the `id` prop is not passed and the `input` id is set to the guid, so we are accounting for this scenario too. 

You should be able to do this by mocking the `guid` helper in the file scope like so:
```ts
import guid from "../../__internal__/utils/helpers/guid";

jest.mock("../../__internal__/utils/helpers/guid");
(guid as jest.MockedFunction<typeof guid>).mockReturnValue("guid-12345");
```